lib/liquid/variable.rb

```diff
@@ -95,17 +95,16 @@ def render(context)
 
     def parse_filter_expressions(filter_name, unparsed_args)
       filter_args = []
-      keyword_args = {}
+      keyword_args = nil
       unparsed_args.each do |a|
         if matches = a.match(JustTagAttributes)
+          keyword_args = {} unless keyword_args
           keyword_args[matches[1]] = Expression.parse(matches[2])
         else
           filter_args << Expression.parse(a)
         end
       end
-      result = [filter_name, filter_args]
-      result << keyword_args unless keyword_args.empty?
-      result
+      [filter_name, filter_args, keyword_args].compact
```

Comment:
Isn't this part slower than before?

Reply:
Replaced `compact` with the old `if` approach

```
       user     system      total        real
pre alloc w kwargs  0.840000   0.010000   0.850000 (  0.842502)
allocated_objects: 2600059
deferred alloc w kwargs  0.800000   0.000000   0.800000 (  0.804708)
allocated_objects: 2600050
pre alloc wo kwargs  0.030000   0.000000   0.030000 (  0.028275)
allocated_objects: 500050
deferred alloc wo kwargs  0.030000   0.000000   0.030000 (  0.030198)
allocated_objects: 400049
pre alloc w arg  0.220000   0.010000   0.230000 (  0.224994)
allocated_objects: 1100050
deferred alloc w arg  0.220000   0.000000   0.220000 (  0.216734)
allocated_objects: 1000050
```

Reply:
We don't need `if` or `compact`, let's just let the value be nil. Either way it's a 40 byte object without extra heap.

Reply:
> We don't need if or compact, let's just let the value be nil.

```
  1) Failure:
VariableUnitTest#test_lax_filter_argument_parsing [/Users/richardmonette/src/github.com/Shopify/liquid/test/unit/variable_unit_test.rb:203]:
--- expected
+++ actual
@@ -1 +1 @@
-[["pluralize", ["comment", "comments"]]]
+[["pluralize", ["comment", "comments"], nil]]


  2) Failure:
VariableUnitTest#test_string_to_filter [/Users/richardmonette/src/github.com/Shopify/liquid/test/unit/variable_unit_test.rb:150]:
Expected: [["image", ["med"]]]
  Actual: [["image", ["med"], nil]]

  3) Failure:
VariableUnitTest#test_filters_without_whitespace [/Users/richardmonette/src/github.com/Shopify/liquid/test/unit/variable_unit_test.rb:130]:
--- expected
+++ actual
@@ -1 +1 @@
-[["textileze", []], ["paragraph", []]]
+[["textileze", [], nil], ["paragraph", [], nil]]


  4) Failure:
VariableUnitTest#test_filter_with_date_parameter [/Users/richardmonette/src/github.com/Shopify/liquid/test/unit/variable_unit_test.rb:124]:
Expected: [["date", ["%m/%d/%Y"]]]
  Actual: [["date", ["%m/%d/%Y"], nil]]

  5) Failure:
VariableUnitTest#test_symbol [/Users/richardmonette/src/github.com/Shopify/liquid/test/unit/variable_unit_test.rb:144]:
Expected: [["image", ["med"]]]
  Actual: [["image", ["med"], nil]]

  6) Failure:
VariableUnitTest#test_filters [/Users/richardmonette/src/github.com/Shopify/liquid/test/unit/variable_unit_test.rb:82]:
Expected: [["textileze", []]]
  Actual: [["textileze", [], nil]]
```

We have tests which assume the result array 2 element, without `nil`. 

I guess we could change that, but I was hoping to avoid modifying any assumptions made by our test suite for this minor perf optimization.

Reply:
This is all internal stuff anyway, it's mostly about clarity and I would just change the tests. Like you say doesn't really matter too much either way, so feel free to do this or not and merge without further review.